### ISSUE TITLE
Private github repository FROM: github.com/org/repo+step suddenly stopped working

### ISSUE THREAD

**Issue by bkendzior** [earthly/earthly#1421]
Builds have suddenly stopped working sometime over the past few days. This is affecting our CI runners, as well as our local builds. Seems like `github.com` URLs are not being rewritten properly all of the sudden?

Earthly version: `earthly version v0.5.22 linux/amd64; Ubuntu 20.04`
earthly buildkitd image: `earthly/buildkitd:v0.5.22`

## Reproduction steps:

Earthfile:
```
FROM github.com/bkendzior/private+base
```

Error
```
$ earthly -V +base                          
           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)
           buildkitd | Comparing running container image ("sha256:e87ea162ed9860b45d111261bfd5839da4ae460cb645079ab109193955f53a77\n") with available image ("sha256:e87ea162ed9860b45d111261bfd5839da4ae46
0cb645079ab109193955f53a77\n")
           buildkitd | Settings do not match. Restarting buildkit daemon with updated settings...
           buildkitd | ...Done
       internal-term | connecting to shellrepeater on 127.0.0.1:8373
            internal | --> docker-image://docker.io/alpine/git:v2.30.1
            internal | --> GIT CLONE https://github.com/bkendzior/private.git
            internal | [          ] resolve docker.io/alpine/git:v2.30.1 ... 0%
            internal | hint: Using 'master' as the name for the initial branch. This default branch name
            internal | hint: is subject to change. To configure the initial branch name to use in all
            internal | hint: of your new repositories, which will suppress this warning, call:
            internal | hint: 
            internal | hint:    git config --global init.defaultBranch <name>
            internal | hint: 
            internal | hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
            internal | hint: 'development'. The just-created branch can be renamed via this command:
            internal | hint: 
            internal | hint:    git branch -m <name>
            internal | Initialized empty Git repository in /tmp/earthly/buildkit/runc-overlayfs/snapshots/snapshots/3094/fs/
            internal | fatal: could not read Username for 'https://github.com': terminal prompts disabled
            internal | WARN: (GIT CLONE https://github.com/bkendzior/private.git) error fetching default branch for repository https://github.com/bkendzior/private.git: exit status 128
            internal | Completed in 203.785145ms
            internal | [██████████] resolve docker.io/alpine/git:v2.30.1 ... 100%
            internal | WARN: Canceled
            internal | Completed in 203.962622ms
Summary of timing information
Note that the times do not include the expansion of commands like BUILD, FROM, COPY (artifact).
            internal | () 407.747767ms
===============================================================
Total           407.747767ms
Total (real)    947.129888ms
Error stack trace:
github.com/moby/buildkit/util/stack.Enable
        /go/pkg/mod/github.com/earthly/buildkit@v0.7.1-0.20210728003253-199ad6a5c213/util/stack/stack.go:77
github.com/moby/buildkit/util/grpcerrors.FromGRPC
        /go/pkg/mod/github.com/earthly/buildkit@v0.7.1-0.20210728003253-199ad6a5c213/util/grpcerrors/grpcerrors.go:188
github.com/moby/buildkit/util/grpcerrors.UnaryClientInterceptor
        /go/pkg/mod/github.com/earthly/buildkit@v0.7.1-0.20210728003253-199ad6a5c213/util/grpcerrors/intercept.go:41
google.golang.org/grpc.(*ClientConn).Invoke
        /go/pkg/mod/google.golang.org/grpc@v1.38.0/call.go:35
github.com/moby/buildkit/api/services/control.(*controlClient).Solve
        /go/pkg/mod/github.com/earthly/buildkit@v0.7.1-0.20210728003253-199ad6a5c213/api/services/control/control.pb.go:1321
github.com/moby/buildkit/client.(*Client).solve.func2
        /go/pkg/mod/github.com/earthly/buildkit@v0.7.1-0.20210728003253-199ad6a5c213/client/solve.go:215
golang.org/x/sync/errgroup.(*Group).Go.func1
        /go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1371
Error: build target: build main: bkClient.Build: failed to solve: failed to load cache key: error fetching default branch for repository https://github.com/bkendzior/private.git: exit status 128
Warning: timedout while sending analytics
```

## Fix:

Adding the following to my ~/.earthly/config.yaml has the build running as expected:

```
git:
    github.com:
        pattern: 'github.com/([^/]+)/([^/]+)'
        substitute: 'ssh://git@github.com:22/$1/$2.git'
        auth: ssh
```

Previously no ~/.earthly/config.yaml was needed; the default (`auth: auto`?) with a running ssh-agent & a key loaded was enough.

**Comment by andrewsykim** [comment]
@bkendzior just some clarifying questions:

* Was the earthly version updated when you saw this behavior change or was this all using v0.5.22?
* Was the default branch of your git repo changed recently?

**Comment by bkendzior** [comment]
@andrewsykim - no to both; I was also able to reproduce this by manually specifying an older `-buildkit-image` (0.5.18). If I manually specify the branch; the build fails with:

`Error: build target: build main: bkClient.Build: failed to solve: failed to load cache key: failed to fetch remote https://github.com/rios-ai/earth-to-rios.git: exit status 128`

**Comment by bkendzior** [comment]
> Was the `SSH_AUTH_SOCK` environment variable in your setup changed recently? I think we have some auto detection logic by checking it.

Not to my knowledge - it's properly set and works fine for normal `git clone`'ing. I also attempted to set `export EARTHLY_SSH_AUTH_SOCK=$SSH_AUTH_SOCK` and that didn't seem to help.

**Comment by bkendzior** [comment]
Additionally; I've seen this error: https://github.com/earthly/earthly/blame/17feb937be8f295f0e094a09b6999c82ad3a6a40/buildcontext/gitlookup.go#L170 pop up a couple times when this problem is occurring.

**Comment by alexcb** [comment]
#1461 fixes this issue.

**Comment by alexcb** [comment]
This has been fixed and will be released under v0.6.2
